src/openloco/townmgr.cpp

```diff
@@ -46,6 +47,96 @@ namespace openloco::townmgr
     // 0x0049748C
     void update_monthly()
     {
-        call(0x0049748C);
+        for (town& currTown : towns())
+        {
+            if (currTown.name == string_ids::null)
+            {
+                continue;
+            }
+
+            // scroll history
+            if (currTown.history_size == 12 * 20)
+            {
+                memmove(&currTown.history, &currTown.history + 1, 12 * 20 - 1);
+            }
+            else
+            {
+                currTown.history_size++;
+            }
+
+            int32_t newPop = std::max(currTown.population - currTown.unk_14C, 0);
+            int eax = newPop / 50;
+
+            int edx = 0;
+            while (eax > 0xFF)
+            {
+                eax -= 20;
+                edx += 1000;
+            }
+
+            if (edx != 0)
+            {
+                currTown.unk_14C += edx;
+                int eax2 = edx / 50;
+
+                for (uint8_t i = 0; i < currTown.history_size; i++)
+                {
+                    currTown.history[i] = std::max(currTown.history[i] - (eax2 & 0xFF), 0);
+                }
+            }
+
+            currTown.history[currTown.history_size - 1] = eax & 0xFF;
+            eax &= ~0xFF; // al = 0;
+            uint8_t al = 0;
+
+            uint8_t max = 0;
+            for (int i = 0; i < currTown.history_size; i++)
+            {
+                max = std::max(max, currTown.history[i]);
+            }
+
+            edx = currTown.unk_14C;
+            while (al <= 0xEB && edx > 0)
+            {
+                al += 20;
+                edx -= 1000;
+            }
+
+            edx -= currTown.unk_14C;
+            if (edx != 0)
+            {
+                edx = -edx;
+                currTown.unk_14C -= edx;
+                edx /= 50;
+
+                for (int i = 0; i < currTown.history_size; i++)
+                {
+                    currTown.history[i] += edx;
+                }
+            }
+
+            int16_t ax = -1;
+            uint32_t ebx = currTown.unk_198;
+            while (ebx != 0)
+            {
+                int ecx;
+                for (ecx = 0; ecx < 32; ecx++)
+                {
+                    if ((ebx & (1 << ecx)) != 0)
+                    {
+                        ebx &= ~(1 << ecx);
+                        break;
+                    }
+                }
+
+                ax = std::max(ax, currTown.unk_158[ecx]);
+            }
+
+            currTown.build_speed = std::clamp(ax / 100, 1, 3);
```

Comment:
I don't think this code works as the original (show in table).

ax | build_speed
-- | --
99 | 1
100 | 2
199 | 2
200 | 3
299 | 3
300 | 4